### PR TITLE
Update autocomplete component to support custom template

### PR DESCRIPTION
**Pull request by alex-ju**
This PR:
- adds a new option for the autocomplete to support custom template
- add the custom template version of autocomplete to contacts

👉 [Preview component](https://content-publisher-revie-pr-571.herokuapp.com/component-guide/autocomplete/autocomplete_with_custom_template)

[Trello card](https://trello.com/c/05U8l7DF)

### CODE REVIEW

File: app/assets/javascripts/components/autocomplete.js [hunk 1]
```diff
@@ -40,4 +40,37 @@ if ($multiselects) {
...
+    defaultValue: '',
+    templates: {
+      inputValue: function (result) {
+        if (result) {
+          return result.split(' - ')[0]
+        }
+      },
+      suggestion: function (result) {
+        if (result) {
+          var resultItems = result.split(' - ')
```
Comment (benthorner) [inline app/assets/javascripts/components/autocomplete.js]: Are we sure this character sequence won't occur in what should be the first entry?

File: app/assets/javascripts/components/autocomplete.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ if ($autocompletes) {
     new accessibleAutocomplete.enhanceSelectElement({
       selectElement: $el,
       minLength: 3,
-      showNoOptionsFound: false,
+      defaultValue: '',
```
Comment (benthorner) [inline app/assets/javascripts/components/autocomplete.js]: Not sure why we're changing this?

File: app/views/components/docs/autocomplete.yml [hunk 3]
```diff
@@ -80,3 +94,24 @@ examples:
       selected_options:
           - fr
           - de
+  autocomplete_with_custom_template:
```
Comment (benthorner) [inline app/views/components/docs/autocomplete.yml]: This one doesn't seem to work in preview :-(

File: app/assets/javascripts/components/autocomplete.js (hunk 1 above)
Comment (alex-ju) [reply app/assets/javascripts/components/autocomplete.js]: Yes, we tested and for now it's safe to use it. It's only present in one organisation which doesn't have any contacts.

File: app/views/components/docs/autocomplete.yml (hunk 3 above)
Comment (alex-ju) [reply app/views/components/docs/autocomplete.yml]: Works here https://content-publisher-revie-pr-571.herokuapp.com/component-guide/autocomplete/autocomplete_with_custom_template if you type `Uni` for example

File: app/assets/javascripts/components/autocomplete.js (hunk 2 above)
Comment (alex-ju) [reply app/assets/javascripts/components/autocomplete.js]: We want to show if there're no results found and we want to set the default for this field to an empty option. These changed make the preview of the components work as expected - which didn't until now.

**Comment by alex-ju** [comment]
@benthorner I'm happy to close this in favour of other approaches. Did the best I could given the way the data is presented in the HTML.

**Comment by benthorner** [comment]
@alex-ju sorry - it's a tricky problem. That empty organisation could potentially have contacts in future, so this doesn't seem like a net improvement as it would introduce a waiting bug. We do currently have the same presentation style for the ministerial roles, which seemed to be OK in user testing. Perhaps, as you say, we could leave it as it is for the time being.

File: app/views/components/docs/autocomplete.yml (hunk 3 above)
Comment (benthorner) [reply app/views/components/docs/autocomplete.yml]: Oops - me being dim.

**Pull request closed without merging**